latex/definitions.tex

```diff
@@ -0,0 +1,66 @@
+\section{Miscellaneous Definitions}
+\label{sec:misc-defs}
+To keep this document fairly self-contained, this section collects definitions of
+many of the technical terms we use in presenting the Cardano ledger specification.
+Some online references for this section are
+\cite{www-essential-cardano-glossary,www-developers-cardano-reintro,www-ledger-academy-glossary}.
+
+
+\subsection{Cardano Time Handling}
+\label{sec:cardano-time-handling}
+Some online references for this subsection are
+\cite{www-docs-cardano-network,www-docs-cardano-time,www-iohk-blog-ouroboros,www-ledger-academy-stake-ada}.
+
+In Cardano, the Ouroboros proof-of-stake (PoS) consensus protocol models the passage
+of physical time as an infinite sequence of time slots and epochs.
+
+\begin{definition}[slot]
+  A \emph{slot} is a discrete time interval in which a single block may be produced; it
+  is the fundamental time unit within the blockchain's consensus protocol.
+\end{definition}
+Slots should be long enough for a new block to have a good chance to reach
+the next slot leader in time.  For example, the slot length in the Byron era was 20
+seconds, while in Shelley and later eras it is 1 second.
+
+Not every slot results in a new block.  Indeed, in any given slot, one or more
+block-producing nodes are nominated (probabilistically based on stake distribution)
+to be \textit{slot leaders} and given the opportunity to produce a new block.
+For example, in Shelley and later eras, on average only 0.05 of slots will produce a
+block (resulting in 20-second intervals between blocks).
+
+\begin{definition}[epoch]
+  An \emph{epoch} is a period of time, containing some number of slots, used to select
+  block-producing nodes.
+\end{definition}
+For example, in Shelley and later eras, an epoch consists of roughly 432,000 slots (or five
+days if we assume a slot length of 1 second).
+
+\begin{definition}[slot number]
+  \emph{Slot number} may refer to a slot's position within the current epoch or it
+  may mean the absolute slot count since the genesis block.  The context should make
```

Comment:
"genesis block" is used here as a unit of time even though its name doesn't suggest it is. Maybe explain briefly?

Reply:
Actually, "since the genesis block" seems clear enough to me; I believe it simply means the time elapsed since the genesis block was created.

Reply:
...but you're right, we might as well add a definition for "genesis block."